ui/ui-components/hooks/useApi.ts

```diff
@@ -1,14 +1,12 @@
+import { NextPageContext } from "next";
```

Comment:
```suggestion
import type { NextPageContext } from "next";
```
It _might_ be worth using a type import. [Details here](https://stackoverflow.com/a/64243357)